arch/arm/src/samv7/sam_ssc.c

```diff
@@ -3053,23 +3035,15 @@ static int ssc_dma_allocate(struct sam_ssc_s *priv)
 static void ssc_dma_free(struct sam_ssc_s *priv)
 {
 #ifdef SSC_HAVE_TX
-  if (priv->tx.dog)
-    {
-       wd_delete(priv->tx.dog);
-    }
-
+  wd_cancel(&priv->tx.dog);
```

Comment:
Are all instances of struct wdog_s properly zeroed?  I think there would be a crash if wd_cancel() were called without at least one wd_start() if the struct wdog_s is not zeroed(?)

Reply:
This is a general question and applies to all places where wd_cancel() is called, not just the one that I have flagged here.

Reply:
The usage can be classified as:
1.The first catagory contain both work_s and wdog_s which is safe because work_s already has the same requirement.
2.The second catagory declare the global variable which will zero by default.
3.The third catagory is dynamically allocated struct which either done by kmm_malloc/memset or kmm_zalloc.
But after review each wdog_s instance, I found one driver in catagory 3 but don't zero memory, here is PR: https://github.com/apache/incubator-nuttx/pull/1564